### DISCUSSION TITLE
MySQL Support?

### DISCUSSION THREAD

**Discussion by fahadshery** [mealie-recipes/mealie#707] [Feature Request]
**Describe the bug**
I came across this project and instantly loved it! many thanks for your efforts!
Not a bug but a simple question if this project supports `MySQL` database as the backend? 
The reason is that I have already setup a `MySQL` instance hosting `nextcloud`, `nginx proxy` data etc.
It would be nice if I could store `mealie data` in there so that everything is in **one place**.

thanks once again!

**Comment by hay-kot** [comment]
There are no plans for MySQL support. Postgres is still in beta support levels and getting that to working along with maintaining a cross-database test suite is already a lot of work. I'm not looking at adding more complexity to the development workflow.

**Comment by Vashidu** [comment]
Would really like to see a mysql/mariadb support option, I've all but removed PostgreSQL from my docker hosts as it was becoming an issue with every stack needing a postgres container.. moved back to a more traditional mariadb/galera/keepalived/haproxy setup for SQL clustering..

**Comment by hay-kot** [comment]
in PR #725 I've rewritten how the database providers are loaded and it's not much more modular. As such, registering another database that works with SQLAlchemy should be trivial. 

Once PR #725 is merged I will accept PRs that include support for additional databases, provided they include some way to test them in an automated fashion.

**Reply by Vashidu** [reply]
i look forward to making a MySQL module, automated testing shall prove interesting I believe

**Reply by hay-kot** [reply]
PR has been merged

- Checkout [this workflow](https://github.com/hay-kot/mealie/blob/mealie-next/.github/workflows/backend-tests.yml) for how I've setup Postgres testing in the CI/CD pipeline
- Checkout [db_providers.py](https://github.com/hay-kot/mealie/blob/mealie-next/mealie/core/settings/db_providers.py) for how to register a provider. 

Also be sure to include documentation on what's required to setup MySQL if you end up putting a PR together.

**Reply by BobWs** [reply]
So any progress on the mysql/mariadb implementation?

**Reply by hay-kot** [reply]
@BobWs, Postgres and SQLite are still the only supported databases.

**Comment by VitoFe** [comment]
Adding MariaDB as a supported database would provide users with more flexibility in their database choices and could potentially increase the adoption of this project. Users who already have a SQL server (like OP) would not need to install PostgreSQL specifically for this project, which would simplify the setup process.

Moreover, adding MariaDB support should not add significant overhead to the current implementation. This is because MariaDB and SQLite share many similarities in terms of SQL syntax and database management. Both are based on the SQL language and use similar data types and commands. Therefore, much of the existing code used for SQLite could be reused or slightly modified for MariaDB, minimizing the changes needed in the codebase.

I believe that adding MySQL/MariaDB support would be a great enhancement to the project and would be appreciated by many users.

**Reply by boc-the-git** [reply]
> I believe that adding MySQL/MariaDB support would be a great enhancement to the project and would be appreciated by many users.

I think the fact this discussion is almost 3 years old with limited activity is an indication it isn't a huge thing. If someone wants to raise a PR to implement it that doesn't in any way increase the support burden of the project, I'm sure we'd be happy to merge it. @hay-kot said as much [in 2021](https://github.com/mealie-recipes/mealie/discussions/707#discussioncomment-1439413)!

I don't see any of the project contributors looking at this.

**Reply by Kuchenpirat** [reply]
Yeah, am with @boc-the-git on this one. 

I think if you are not willing to have a dedicated database container, just going with SQLite is working out great as it has pretty low overhead.
The advatage of going with the Postgres container is the support for fuzzysearch which will not be included with a simple MariaDB add. So i don't think it is worth the additional complexity.
